frida_tools/tracer.py

```diff
@@ -581,6 +609,8 @@ def _create_trace_script(self):
 var pending = [];
 var timer = null;
 
+state.userdata = "%(userdata)s";
+
 rpc.exports = {
```

Comment:
Hmm, perhaps it would be cleaner to define an RPC export here named `init`, having the same signature as the described in the [Gadget docs](https://frida.re/docs/gadget/). This way we can call it right after loading the script, and there is no need to escape anything.

Reply:
This was a challenging request, as I had to understand the concept of Python RPC exports etc., but I believe I got it in the end.  I have removed all escaping of `options.userdata`, inserting the following init() function in `rpc.exports`:

```
rpc.exports = {
    init: function (stage, parameters) {
        state.userdata = %{userdata}s;
    },
    dispose: function () {
        flush();
    },
```

Your idea is clever and clean, bypassing the need to be worried about the leading and trailing double quotes.  Please verify that it is what you meant.

Reply:
Glad to hear. I've clarified a bit in the other comment, hope it makes it clearer. (Sorry for being so brief earlier.)